src/MessageHandler/LostMessagesConsumer.php

```diff
@@ -44,36 +44,57 @@ public function setMessageTtl(int $messageTtl): void
     /**
      * Requeue message with date superior to message ttl
      * Then max retry reached, message is put in dead letter list
+     *
+     * @param int|null $secondTime
      */
-    public function requeueOldMessages()
+    public function requeueOldMessages(?int $secondTime = null): void
     {
-        array_map(
-            [$this, 'requeueOldMessageFromList'],
-            iterator_to_array($this->queue->getWorkingLists($this->redisClient))
-        );
+        $restTime = $secondTime - (time() - $secondTime);
+
+        shuffle($workingLists = iterator_to_array($this->queue->getWorkingLists($this->redisClient)));
+
+        foreach($workingLists as $workingList)
+        {
+            if ($restTime <= 0 ){
+                break;
+            }
+
+            $this->requeueOldMessageFromList($workingList, $restTime);
+        }
     }
 
-    protected function requeueOldMessageFromList(string $list)
+    /**
+     * Requeue old messages from $list, and remove them if
+     * $messageTTL or time is exceed.
+     *
+     * @param string   $list
+     * @param int $time
+     */
+    protected function requeueOldMessageFromList(string $list, int $time): void
     {
-        for ($i = 0, $n = $this->redisClient->llen($list); $i < $n; $i++) {
-            $serializedMessage = $this->redisClient->rpoplpush($list, $list);
+        $maxMessages = $this->redisClient->llen($list);
 
-            // check if message is old enough
-            $message = MessageEnvelope::unserializeMessage($serializedMessage);
-            if ((time() - $message->getUpdatedAt()->format('U')) > $this->messageTtl
-                && $this->redisClient->lrem($list, 0, $serializedMessage)) {
-                if ($message->getRetry() < $this->maxRetry) {
-                    $this->addMessageInQueueList($message);
+        while (($maxMessages-- > 0)  && ($serializedMessage = $this->redisClient->rpoplpush($list, $list))) {
+            $message = empty($serializedMessage) ? null : MessageEnvelope::unserializeMessage($serializedMessage);
 
-                    continue;
-                }
+            if (!empty($message)) {
+                // Check if message is old enough
+                if (($time - $message->getUpdatedAt()->format('U')) > $this->messageTtl && $this->redisClient->lrem($list, 0, $serializedMessage)) {
```

Comment:
time ??? $time is the max treatment time, nothing to compare the updateAt if the message ??